content/v2.0/application/booting.md

```diff
@@ -0,0 +1,164 @@
+---
+title: Booting
+order: 40
+---
+
+Hanami applications support a **prepared** state and a **booted** state.
+
+Whether your app is prepared or booted determines whether its components are **lazily loaded** on demand (prepared stated), or **eagerly loaded** up front (booted state).
+
+This difference may not sound like much, but the ability of a prepared app to load _just enough_ of the app to undertake a particular task can be a huge performance advantage when it comes to testing, worker processes and other use cases where your app doesn't need every component to perform the job at hand.
+
+Conversely, a booted app is fully initialized, with every component ready to do its work. This is ideal whenever you want to incur initialization costs at boot time, such as when serving web requests.
+
+## Hanami.prepare
+
+When you call `Hanami.prepare` (or use `require "hanami/prepare"`), Hanami will prepare your app for use.
+
+This process includes defining various Ruby constants required for your app to function, making the Ruby source files in `app/` autoloadable, and making your components (the classes you've added to the `app/` folder) lazily loadable.
```

Comment:
"defining various Ruby constants required for your app to function" sounds too mysterious for my liking.

Could we make it a bit more specific (without going into too much detail)?

For example:

_tried to write an example and failed_

Perhaps we can just ditch it and instead focus on the impact on the user's own code? e.g.

```suggestion
This process makes the Ruby source files in `app/` autoloadable and makes your components (from the same classes defined in `app/`) lazily loadable.
```

Reply:
Yeah, that's heaps clearer, thanks.